langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/ToJson.java

```diff
@@ -0,0 +1,169 @@
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langlib.value;
+
+import org.ballerinalang.jvm.JSONUtils;
+import org.ballerinalang.jvm.StringUtils;
+import org.ballerinalang.jvm.TypeChecker;
+import org.ballerinalang.jvm.TypeConverter;
+import org.ballerinalang.jvm.commons.TypeValuePair;
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.types.BArrayType;
+import org.ballerinalang.jvm.types.BMapType;
+import org.ballerinalang.jvm.types.BType;
+import org.ballerinalang.jvm.types.BTypes;
+import org.ballerinalang.jvm.types.TypeTags;
+import org.ballerinalang.jvm.util.exceptions.BLangExceptionHelper;
+import org.ballerinalang.jvm.util.exceptions.BallerinaException;
+import org.ballerinalang.jvm.util.exceptions.RuntimeErrors;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.ArrayValueImpl;
+import org.ballerinalang.jvm.values.ErrorValue;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.MapValueImpl;
+import org.ballerinalang.jvm.values.RefValue;
+import org.ballerinalang.jvm.values.TableValueImpl;
+import org.ballerinalang.jvm.values.api.BString;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinalang.jvm.BallerinaErrors.createError;
+import static org.ballerinalang.jvm.util.exceptions.BallerinaErrorReasons.VALUE_LANG_LIB_CONVERSION_ERROR;
+import static org.ballerinalang.jvm.util.exceptions.BallerinaErrorReasons.VALUE_LANG_LIB_CYCLIC_VALUE_REFERENCE_ERROR;
+import static org.ballerinalang.jvm.util.exceptions.RuntimeErrors.INCOMPATIBLE_CONVERT_OPERATION;
+import static org.ballerinalang.util.BLangCompilerConstants.VALUE_VERSION;
+
+/**
+ * Extern function lang.values:toJson.
+ * Converts a value of type `anydata` to `json`.
+ *
+ * @since 2.0
+ */
+@BallerinaFunction(
+        orgName = "ballerina",
+        packageName = "lang.value", version = VALUE_VERSION,
+        functionName = "toJson",
+        isPublic = true
+)
+public class ToJson {
+
+    public static Object toJson(Strand strand, Object value) {
+        try {
+            return convert(value, new ArrayList<>(), strand);
+        } catch (Exception e) {
+            return e;
+        }
+    }
+
+    private static Object convert(Object value, List<TypeValuePair> unresolvedValues, Strand strand) {
+        BType jsonType = BTypes.typeJSON;
+
+        if (value == null) {
+            return null;
+        }
+
+        BType sourceType = TypeChecker.getType(value);
+
+        if (sourceType.getTag() <= TypeTags.BOOLEAN_TAG) {
+            if (TypeChecker.checkIsType(value, jsonType)) {
+                return value;
+            } else {
+                // Has to be a numeric conversion.
```

Comment:
Do we ever reach here? All the numeric basic types are both `anydata` and `json`. So, I don't think you'll ever have to do a numeric conversion?

Reply:
Got it